git-sidecar/rootfs/clone.sh

```diff
@@ -1,23 +1,35 @@
 #!/bin/sh
 set -euo pipefail
+
 set -x
 
-: "${VCS_LOCAL_PATH:=/src}"
-: "${VCS_REPO:?}"
-: "${VCS_REVISION:?}"
-: "${INIT_GIT_SUBMODULES:?}"
+# The Git reference. Required.
+#
+# Also can be a full hex object name.
+# master, v0.1.0, refs/pulls/1/head, 589e15029e1e44dee48de4800daf1f78e64287c0
+: "${BRIGADE_COMMIT_REF:?}"
+
+# The Git SHA1 of the revision.
+: "${BRIGADE_COMMIT_ID:=}"
```

Comment:
After spending some time on reviewing this PR, I realized the clear distinctions of "commit id" vs "commit ref" vs "revision" and concise comments on these.

Even though I still have some nits/confusion between "commit" and "revision", merging this PR will indeed be a significant step forward to make brigade a more general solution as a CI system.

Thank you very much for your efforts 👍